### PR TITLE
fix: support deployment of an Angular app using legacy browser builder

### PR DESCRIPTION
**Pull request by 9kubczas4**
### Description

This change fixes:
- https://github.com/firebase/firebase-tools/issues/6908
- https://github.com/firebase/firebase-tools/issues/7029

The fix was done in 2 areas of the code:
- `@angular-devkit/build-angular:browser (app:build:production) is not a recognized builder` was thrown because `@angular-devkit/build-angular:browser` wasn't recognized as build target.
- `@angular-devkit/build-angular:browser` doesn't create `browser` subfolder in the output path

### Scenarios Tested

I've built and deployed to firebase Angular apps (v17) which use:
- `@angular-devkit/build-angular:browser`
- `@angular-devkit/build-angular:application`

### CODE REVIEW

File: src/frameworks/angular/utils.ts [hunk 1]
```diff
@@ -420,7 +421,13 @@ export async function getBrowserConfig(sourceDir: string, configuration: string)
   );
   const targetOptions = await architectHost.getOptionsForTarget(buildOrBrowserTarget);
   assertIsString(targetOptions?.outputPath);
-  const outputPath = join(targetOptions.outputPath, buildTarget ? "browser" : "");
+
+  const builderName = await architectHost.getBuilderNameForTarget(buildOrBrowserTarget);
```
Comment (leoortizz) [inline src/frameworks/angular/utils.ts:425]: we could `Promise.all` all the promises after line 413

File: src/frameworks/angular/utils.ts [hunk 2]
```diff
@@ -420,7 +421,13 @@ export async function getBrowserConfig(sourceDir: string, configuration: string)
...
+
+  const outputPath =
+    builderName === ExpectedBuilder.APPLICATION
+      ? join(targetOptions.outputPath, buildTarget ? "browser" : "")
+      : targetOptions.outputPath;
```
Comment (leoortizz) [inline src/frameworks/angular/utils.ts:430]: I think we can keep only one ternary here:
```suggestion
  const outputPath = join(
    targetOptions.outputPath,
    buildTarget && builderName !== ExpectedBuilder.APPLICATION ? "browser" : "",
  );
```

Comment (leoortizz) [reply src/frameworks/angular/utils.ts]: oops, right

**Review by jamesdaniels — APPROVED** [review]

**Comment by Erkam246** [comment]
@9kubczas4 could you provide us the configuration for @angular-devkit/build-angular:application

Im getting:

`Error: @angular-devkit/build-angular:application (app:build:production) is not a recognized builder. Please check your angular.json`

with this configuration:

```
"deploy": {
          "builder": "@angular/fire:deploy",
          "options": {
            "version": 2,
            "browserTarget": "app:build:production"
          }
        }
```

**Comment by rinebob** [comment]
Suggestion from @Erkam246 worked for me.  I was trying to start the emulators and getting


c:\aa\projects\friendly-chat>firebase emulators:start
i  emulators: Starting emulators: auth, functions, firestore, hosting, storage
...
i  emulators: Shutting down emulators.
i  hub: Stopping emulator hub

Error: @angular-devkit/build-angular:application (friendlychat:build:production) is not a recognized builder. Please check your angular.json

Removing the deploy configuration from angular.json allowed the emulators to start properly

**Comment by deanbiltup** [comment]
firebase deploy works but the cloud builder fails until I modify to application and then my local no longer works :(.   Also, when I get cloud building it doesn't seem to use my changes at all.  It says success but the app never changes from my last firebase deploy :( :( .  If I leave browser builder, cloud builder fails.

**Comment by 9kubczas4** [comment]
Hi @deanbiltup,

Could you provide me a little more details?
Steps to reproduce, logs etc.

Thank you in advance

**Comment by deanbiltup** [comment]
send me a github user id and I can add you to my github repo since it is
just a demo.    I can shut it down later as I put it all in a temp project
to test out chat.

**Pull request merged**